tests/lib/rules/id-length.js

```diff
@@ -564,6 +676,150 @@ ruleTester.run("id-length", rule, {
+        },
+        {
+            code: "(𐌘) => { 𐌘 * 𐌘 };",
+            parserOptions: {
+                ecmaVersion: 6
+            },
+            errors: [
+                tooShortError
+            ]
+        },
+        {
+            code: "class 𠮟 { }",
+            parserOptions: {
+                ecmaVersion: 6
+            },
+            errors: [
+                tooShortError
+            ]
+        },
+        {
+            code: "class Foo { 𐌘() {} }",
+            parserOptions: {
+                ecmaVersion: 6
+            },
+            errors: [
+                tooShortError
+            ]
+        },
+        {
+            code: "class Foo1 { #𐌘() {} }",
+            parserOptions: {
+                ecmaVersion: 2022
+            },
+            errors: [
+                tooShortErrorPrivate
+            ]
+        },
+        {
+            code: "class Foo2 { 𐌘 = 1 }",
+            parserOptions: {
+                ecmaVersion: 2022
+            },
+            errors: [
+                tooShortError
+            ]
+        },
+        {
+            code: "class Foo3 { #𐌘 = 1 }",
+            parserOptions: {
+                ecmaVersion: 2022
+            },
+            errors: [
+                tooShortErrorPrivate
+            ]
+        },
+        {
+            code: "function foo1(...𐌘) { }",
+            parserOptions: {
+                ecmaVersion: 6
+            },
+            errors: [
+                tooShortError
+            ]
+        },
+        {
+            code: "function foo([𐌘]) { }",
+            parserOptions: {
+                ecmaVersion: 6
+            },
+            errors: [
+                tooShortError
+            ]
+        },
+        {
+            code: "var [ 𐌘 ] = arr;",
+            parserOptions: {
+                ecmaVersion: 6
+            },
+            errors: [
+                tooShortError
+            ]
+        },
+        {
+            code: "var { prop: [𐌘]} = {};",
+            parserOptions: {
+                ecmaVersion: 6
+            },
+            errors: [
+                tooShortError
+            ]
+        },
+        {
+            code: "function foo({𐌘}) { }",
+            parserOptions: {
+                ecmaVersion: 6
+            },
+            errors: [
+                tooShortError
+            ]
+        },
+        {
+            code: "var { 𐌘 } = {};",
+            parserOptions: {
+                ecmaVersion: 6
+            },
+            errors: [
+                tooShortError
+            ]
+        },
+        {
+            code: "var { prop: 𐌘} = {};",
+            parserOptions: {
+                ecmaVersion: 6
+            },
+            errors: [
+                tooShortError
+            ]
+        },
+        {
+            code: "({ prop: obj.𐌘 } = {});",
+            parserOptions: {
+                ecmaVersion: 6
+            },
+            errors: [
+                tooShortError
+            ]
```

Comment:
I think that in all added test cases the number of graphemes is the same as the number of code points. I tried replacing counting graphemes with counting code points, and all tests are still passing:

```diff
-return splitter.countGraphemes(value);
+return [...value].length;
```

Would it be possible to add a test where the number of graphemes is not the same as the number of code points, to ensure that we're counting graphemes? Otherwise, i.e., if such sequence of characters cannot be a valid [IdentifierName](https://tc39.es/ecma262/#prod-IdentifierName), then maybe we don't need to use `GraphemeSplitter` in this particular rule.

Reply:
Unfortunately, there are IdentifierName that differ in the number of grapheme and code points.

A character containing [Variation Selectors](https://en.wikipedia.org/wiki/Variation_Selectors_(Unicode_block)) are a single grapheme with two code points.

For example, `葛󠄀`(`0x845B` `0xE0100`).

```js
[..."葛󠄀"].length; // 2
const 葛󠄀 = "hogehoge"; // valid
```

[bc16453](https://github.com/eslint/eslint/pull/16321/commits/bc16453ff8e815bbac98457f0640b5744ef96d47)

Reply:
Thanks for the info! In that case, we do need `GraphemeSplitter` for correctness.